### ISSUE TITLE
Have to run 'antigen reset' after changing .zshrc

### ISSUE THREAD

**Issue by benwaffle** [zsh-users/antigen#587]
After changing antigen bundles in `.zshrc`, the changes do not take effect (even in new shells) until I run `antigen reset`.

deleting `~/.antigen/init.zsh*` also works

I'm not sure if this is intentional or a bug. Is there a way to fix this?

**Comment by desyncr** [comment]
Take a look here: https://github.com/zsh-users/antigen/wiki/Troubleshooting#my-bundles-wont-update

It's intended as per performance issues. We'll review it in the future.

**Comment by jguenther** [comment]
The info in that wiki article didn't work for me. Using this config:
```
ANTIGEN_CACHE=$HOME/.antigen/init.zsh
typeset -a ANTIGEN_CHECK_FILES=($HOME/.zshrc $HOME/.someotherfile)
```

after running `antigen reset` and opening a new shell, I always get this error if `ANTIGEN_CHECK_FILES` has more than 1 file (no error if it's just 1 file):
```
-zcache-generate-cache:zcompile:81: can't write zwc file: /path/to/home/.zshrc /path/to/home/.someotherfile.zwc
```

Looked into it briefly and I think it's from [this line](https://github.com/zsh-users/antigen/blob/89b74e54016e880de165f9b0d2fb62e5b7883ddf/src/ext/cache.zsh#L71). 

It's quoting "$ANTIGEN_CHECK_FILES" which means `zcompile` gets all files as 1 arg instead of separately. Since this var is an array, the correct way to do this without breaking on files/dirs with whitespace etc is:
```
zcompile "${ANTIGEN_CHECK_FILES[@]}"
```

This will properly quote each element of the array.

**Comment by desyncr** [comment]
@jguenther You're right. I'll look into this.

**Comment by Ameausoone** [comment]
Same problem 

zsh 5.4.2 (x86_64-apple-darwin16.7.0)

Antigen v2.2.1

When I add new bundles in .antigenrc, I had to delete my ~/.antigen/ directory install new bundle

**Comment by desyncr** [comment]
I'm addressing this issue in https://github.com/zsh-users/antigen/pull/588. Hopefully will be available in 2.2.2.
